app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -82,148 +133,168 @@ const BigDelegatorOffer = () => (
 )
 
 interface Props {
-  updateStakePoolIdentifier: any
-  removeStakePool: any
-  stakePool: any
-  delegationFee: any
-  calculatingDelegationFee: any
+  stakePool: Stakepool
+  delegationFee: Lovelace
+  calculatingDelegationFee: boolean
   delegationValidationError: any
-  isShelleyCompatible: any
-  confirmTransaction: any
-  selectAdaliteStakepool: any
-  shouldShowConfirmTransactionDialog: any
-  txSuccessTab: any
+  isShelleyCompatible: boolean
+  shouldShowConfirmTransactionDialog: boolean
+  txSuccessTab: string
   gettingPoolInfo: boolean
-  poolRecommendation: any
-  pool: any
+  poolRecommendation: PoolRecommendation
+  pool: Stakepool
   isBigDelegator: boolean
+  validStakepools: Stakepools
   withAccordion: boolean
   title: string
+  updateStakePoolIdentifier: (poolHash: string, validationError?: any) => void
+  confirmTransaction: (txConfirmType) => Promise<void>
+  selectAdaliteStakepool: () => void
 }
 
-class Delegate extends Component<Props, {dropShadow: boolean}> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      dropShadow: false,
-    }
-  }
+const Delegate = ({
+  stakePool,
+  delegationFee,
+  calculatingDelegationFee,
+  delegationValidationError,
+  isShelleyCompatible,
+  shouldShowConfirmTransactionDialog,
+  txSuccessTab,
+  gettingPoolInfo,
+  poolRecommendation,
+  pool,
+  isBigDelegator,
+  validStakepools,
+  withAccordion,
+  title,
+  updateStakePoolIdentifier,
+  confirmTransaction,
+  selectAdaliteStakepool,
+}: Props): h.JSX.Element => {
+  useEffect(
+    () => {
+      if (ADALITE_CONFIG.ADALITE_STAKE_POOL_ID !== '') selectAdaliteStakepool()
```

Comment:
I know this code predates this PR but do we really need to check the config here?

I'm checking the logic of `selectAdaliteStakepool()` (which is removed from this config variable by a few layers of abstraction so one is wondering if it actually does what it's supposed to) and it would imho be simpler to refactor/generalize the function to `tryPreselectingRecommendedPool()` and set it if there is actually any available recommendation, otherwise it would do nothing

Reply:
Good point, fixed.